### PR TITLE
Bug 1153888 - [Messages] Contacts.findBy and other Contacts.* functio…

### PR DESCRIPTION
**Pull request by anubhav7495**
…ns should never return null r=azasypkin

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=762767d53e34ab2908238ba38cdd953a71a8276f

File: apps/sms/views/conversation/js/conversation.js [hunk 1]
```diff
@@ -2676,7 +2675,7 @@ var ConversationView = {
       return;
     }
 
-    if (!contacts || !contacts.length) {
```
Comment (azasypkin) [inline apps/sms/views/conversation/js/conversation.js]: Looks like ```ConversationView.searchContact``` can still return ```null```, you need to take care about this method too - make it to return empty array as well.

File: apps/sms/views/conversation/js/information.js [hunk 2]
```diff
@@ -450,7 +450,7 @@ Information.prototype = {
           return;
         }
 
-        var isContact = contacts !== null && !!contacts.length;
+        var isContact = contacts.length > 0;
```
Comment (azasypkin) [inline apps/sms/views/conversation/js/information.js]: nit: please merge this variable with the ```if``` below:
```
if (contacts.length > 0) {
```

File: apps/sms/views/shared/js/activity_handler.js [hunk 3]
```diff
@@ -376,8 +376,8 @@ var ActivityHandler = {
 
         return Contacts.findByAddress(message.sender).then(function(contacts) {
           var sender = message.sender;
-          if (!contacts) {
-            console.error('We got a null contacts for sender:', sender);
+          if (!contacts.length) {
```
Comment (azasypkin) [inline apps/sms/views/shared/js/activity_handler.js]: nit: please replace these ```if``` and ```else if``` just with (there is no value in this ```console.error``` anymore):
```
var sender = contacts.length && contacts[0].name &&
  contacts[0].name.length && contacts[0].name[0] ||
  message.sender;
```

File: apps/sms/views/shared/js/contacts.js [hunk 4]
```diff
@@ -115,9 +115,7 @@
          *    result a no-op.
          *
          */
-        return Promise.resolve(
-          filter.filterValue === undefined ? null : []
-        );
+        return Promise.resolve([]);
```
Comment (azasypkin) [inline apps/sms/views/shared/js/contacts.js]: I still see that we check for ```if(!contacts)``` in this file see ```findByPhoneNumber``` and ```findExact```.

File: apps/sms/views/shared/test/unit/contacts_test.js [hunk 5]
```diff
@@ -789,7 +789,8 @@ suite('Contacts', function(done) {
     // to the console
     test('({}, ...)', function(done) {
```
Comment (azasypkin) [inline apps/sms/views/shared/test/unit/contacts_test.js]: nit: remove the comment above, it's no longer relevant. And it's not still clear why you haven't changed ```assert.ok``` to ```assert.isTrue```?

File: apps/sms/views/conversation/js/conversation.js (hunk 1 above)
Comment (anu7495) [inline apps/sms/views/conversation/js/conversation.js]: I changed the return value to empty array for ConversationView.searchContact
conversation_test.js  & conversation_integration_test.js give no errors after that.
Do i need to check for somethings else?

Comment (anu7495) [inline apps/sms/views/conversation/js/conversation.js]: Or i should proceed with the commit.

Comment (azasypkin) [inline apps/sms/views/conversation/js/conversation.js]: Yeah, it no tests are failing, it's a good sign :)

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e58cc78a01bd31d150120a2b4b8ad4be53ceb658

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8c2581c1565c5772dbbb77fa14d4109f3c75105a

File: apps/sms/views/shared/js/activity_handler.js [hunk 6]
```diff
@@ -375,13 +375,9 @@ var ActivityHandler = {
...
         return Contacts.findByAddress(message.sender).then(function(contacts) {
-          var sender = message.sender;
-          if (!contacts) {
-            console.error('We got a null contacts for sender:', sender);
-          } else if (contacts.length && contacts[0].name &&
-            contacts[0].name.length && contacts[0].name[0]) {
-            sender = contacts[0].name[0];
-          }
+          var sender = contacts.length && contacts[0].name &&
+          contacts[0].name.length && contacts[0].name[0] ||
```
Comment (azasypkin) [inline apps/sms/views/shared/js/activity_handler.js]: nit: please align this with 2 spaces, like:
```
var sender = contacts.length && contacts[0].name &&
  contacts[0].name.length && contacts[0].name[0] ||
  message.sender;
```

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=cbb04e08cb31f18d556adc39690e79750ec24b9e

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2226213f8db81e63e2246883bb19e76763c137b5

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=cbb04e08cb31f18d556adc39690e79750ec24b9e

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=309cd21d17031d593f365b393a237f7b6ead237c

**Pull request merged**